evalml/tests/component_tests/test_nullable_types_replacer.py

```diff
@@ -0,0 +1,136 @@
+            "nullable_integer": [0, 1, 2, 3, None],
+            "non_nullable_age": [20, 21, 22, 23, 24],
+            "nullable_age": [20, None, 22, 23, None],
+            "non_nullable_boolean": [True, False, True, False, True],
+            "nullable_boolean": [None, True, False, True, False],
+        }
+    )
+
+
+@pytest.mark.parametrize("input_type", ["ww", "pandas"])
+def test_replace_nullable_types(nullable_data, input_type):
+    X = nullable_data
+
+    nullable_types_replacer = ReplaceNullableTypes()
+
+    X = X.astype(
+        {
+            "nullable_integer": "Int64",
+            "nullable_age": "Int64",
+            "nullable_boolean": "boolean",
+        }
+    )
+
+    assert str(X.dtypes.loc["non_nullable_integer"]) == "int64"
+    assert str(X.dtypes.loc["nullable_integer"]) == "Int64"
+    assert str(X.dtypes.loc["non_nullable_age"]) == "int64"
+    assert str(X.dtypes.loc["nullable_age"]) == "Int64"
+    assert str(X.dtypes.loc["non_nullable_boolean"]) == "bool"
+    assert str(X.dtypes.loc["nullable_boolean"]) == "boolean"
+
+    if input_type == "ww":
+        X.ww.init(logical_types={"nullable_age": AgeNullable, "non_nullable_age": Age})
+        assert isinstance(X.ww.logical_types["nullable_integer"], IntegerNullable)
+        assert isinstance(X.ww.logical_types["nullable_age"], AgeNullable)
+        assert isinstance(X.ww.logical_types["nullable_boolean"], BooleanNullable)
+
+    nullable_types_replacer.fit(X)
+
+    assert set(nullable_types_replacer._nullable_int_cols) == {
+        "nullable_integer",
+        "nullable_age",
+    }
+    assert nullable_types_replacer._nullable_bool_cols == ["nullable_boolean"]
+
+    X_t, y_t = nullable_types_replacer.transform(X)
+    assert set(X_t.columns) == set(X.columns)
+    assert X_t.shape == X.shape
+
+    # Check the pandas dtypes
+    assert str(X_t.dtypes.loc["non_nullable_integer"]) == "int64"
+    assert str(X_t.dtypes.loc["nullable_integer"]) == "float64"
+    assert str(X_t.dtypes.loc["non_nullable_age"]) == "int64"
+    assert str(X_t.dtypes.loc["nullable_age"]) == "float64"
+    assert str(X_t.dtypes.loc["non_nullable_boolean"]) == "bool"
+    assert str(X_t.dtypes.loc["nullable_boolean"]) == "category"
+
+    # Check the Woodwork dtypes
+    assert isinstance(X_t.ww.logical_types["non_nullable_integer"], Integer)
+    assert isinstance(X_t.ww.logical_types["nullable_integer"], Double)
+    if input_type == "ww":
+        assert isinstance(X_t.ww.logical_types["non_nullable_age"], Age)
+    else:
+        assert isinstance(X_t.ww.logical_types["non_nullable_age"], Integer)
+    assert isinstance(X_t.ww.logical_types["nullable_age"], Double)
+    assert isinstance(X_t.ww.logical_types["non_nullable_boolean"], Boolean)
+    assert isinstance(X_t.ww.logical_types["nullable_boolean"], Categorical)
+
+
+@pytest.mark.parametrize("input_type", ["ww", "pandas"])
+def test_replace_nullable_types_boolean_target(nullable_data, input_type):
```

Comment:
nit-pick: Verify that y_t has a woodwork schema

Reply:
Gucci